### ISSUE TITLE
Can't find Supervised Contrastive Loss.

### ISSUE THREAD

**Issue by RisabBiswas** [kathyliu579/MCSC#2]
Hello Qianying! Thanks for uploading the code. I am trying to find the block where you define the supervised contrastive loss. I see you did an import "from utils.supcon_loss_romdom_seleted import BlockConLoss" in your training code, but could not find the utils file. 

Can you kindly let me know?

**Comment by kathyliu579** [comment]
Hi 

sorry for this issue. I found that this folder was not uploaded. now i upload it. If you have any other problem please let me know.

**Comment by RisabBiswas** [comment]
Thanks a lot for your quick response!! :)  Got the file now. Also, I am using the supervised contrastive loss block as a stand-alone module. Is the below usage correct? Consider the 'features' as the representation at any intermediate stage or the predicted segmentation mask. And labels be the GT.

import torch
from utils import *

### Define the SupConLoss and BlockConLoss
supcon_loss = SupConLoss()

features = torch.rand((8, 3, 32, 32))
labels = torch.randint(0, 2, (8, 1, 32, 32)).long()

### Calculate supervised contrastive loss
loss = supcon_loss(features, labels)
print("Supervised Contrastive Loss:", loss.item())

**Comment by kathyliu579** [comment]
Hi sorry for the late reply. Actually, the input should be (bsz, c, h, w) for feature and  (bsz, h, w) for label. So you donot need four dimensions for labels. and another thing is that the dim of feature is usually be 128 or 256 instead of 3. 
Thanks, hope this can help you

**Comment by RisabBiswas** [comment]
Thanks! @kathyliu579
